### PR TITLE
Draft: updated the switch statement to include more checks

### LINKED ISSUE #321: Wrong notification saying "X assigned you to merge request" when the issue is updated (e.g., new commits pushed)
This seems to be a known issue, but can anyone guide me to resolve this?
It is a bit annoying to get notification every time someone pushes a new commit.
image

### PR DESCRIPTION
**Pull request by tschuyebuhl**
#### Summary

Added a bit more checks in case a MR is updated on Gitlab. Please note that this is still WIP as I'm trying to work my way into the plugin. Would be happy to take any suggestions or feedback. I was also thinking to maybe send a message to all the people that were either assigned or unassigned, but I do not know how feasible that would be.

#### Ticket Link

This fixes #321

### CODE REVIEW

File: server/webhook/merge_request.go [hunk 1]
```diff
@@ -33,8 +33,15 @@ func (w *webhook) handleDMMergeRequest(event *gitlab.MergeEvent) ([]*HandleWebho
...
-			message = fmt.Sprintf("[%s](%s) assigned you to merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+			if event.ObjectAttributes.OldRev != "" {
+				message = fmt.Sprintf("[%s](%s) pushed a new commit to merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+			} else {
+				prevAssignees := event.Changes.Assignees.Previous
+				currAssignees := event.Changes.Assignees.Current
+				if len(prevAssignees) != len(currAssignees) || !compareAssignees(prevAssignees, currAssignees) {
+					message = fmt.Sprintf("[%s](%s) updated the assignees of merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+				}
+			}
```
Comment (mickmister) [inline server/webhook/merge_request.go:44]: Some initial thoughts on this:

Posting that a new commit is on a PR can be a bit noisy, so we want to make sure a given subscription is expecting this kind of message to come in. I personally don't think it's relevant enough to warrant a DM notification.

Also, we would still want to support the "assigned you to merge request" message when a given user is added as an assignee. That's one of the most important kinds of DM notification imo

Comment (tschuyebuhl) [reply server/webhook/merge_request.go]: Updated the code to support messages for removing/adding assignees. It should work fine. 
image

**Comment by tschuyebuhl** [comment]
I noticed that there was another PR fixing the issue in question. Sorry for taking so long with the PR. Should I continue working on this? In any case, I pushed my current work. Maybe it will come useful :). Anyways, it was a real pleasure to work with you guys. It's just awesome for me. I hope that I will get to code some more in this repo.

**Comment by mickmister** [comment]
@tschuyebuhl Indeed this was mistakenly being worked on internally, in parallel with this PR. Sorry about this, I should have noticed this and communicated. I will go ahead and close this in favor of https://github.com/mattermost/mattermost-plugin-gitlab/pull/395

@tschuyebuhl Please feel free to join the community server if you're interested in interacting more with the team https://community.mattermost.com. We have a weekly public developer's meeting on Wednesdays at 11:30am EDT, hosted at https://community.mattermost.com/core/channels/developers-meeting. We encourage you to join that if that sounds interesting.

Thanks for your efforts! I apologize again about the logistics on this issue.

Thanks for supporting Mattermost!

**Pull request closed without merging**